### PR TITLE
Generate reconcilers for resources out of tree.

### PR DESCRIPTION
**Pull request by n3wscott**
Adding a `--force-kinds` flag to the reconciler generator to allow us to generate reconcilers for non-owned types.

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -42,15 +42,17 @@ EXTERNAL_INFORMER_PKG="k8s.io/client-go/informers" \
     k8s.io/client-go \
     k8s.io/api \
     "admissionregistration:v1beta1 apps:v1 autoscaling:v1,v2beta1 batch:v1,v1beta1 core:v1 rbac:v1" \
-    --go-header-file ${REPO_ROOT_DIR}/hack/boilerplate/boilerplate.go.txt
+    --go-header-file ${REPO_ROOT_DIR}/hack/boilerplate/boilerplate.go.txt \
+    --force-kinds "Namespace"
```
Comment (mattmoor) [inline hack/update-codegen.sh]: Can we `s/kinds/genreconciler/` for clarity?

Comment (n3wscott) [reply hack/update-codegen.sh]: sure

**Comment by n3wscott** [comment]
I need https://github.com/knative/pkg/pull/1161

Gonna merge that.

File: codegen/cmd/injection-gen/generators/reconciler_reconciler_stub.go [hunk 2]
```diff
@@ -115,11 +115,13 @@ var _ {{.reconcilerInterface|raw}} = (*Reconciler)(nil)
...
 // ReconcileKind implements Interface.ReconcileKind.
 func (r *Reconciler) ReconcileKind(ctx {{.contextContext|raw}}, o *{{.type|raw}}) {{.reconcilerEvent|raw}} {
-	o.Status.InitializeConditions()
+    // TODO: use this if the resource implements InitializeConditions.
+	// o.Status.InitializeConditions()
 
 	// TODO: add custom reconciliation logic here.
-
-	o.Status.ObservedGeneration = o.Generation
+
```
Comment (mattmoor) [inline codegen/cmd/injection-gen/generators/reconciler_reconciler_stub.go]: Remove trailing whitespace:
```suggestion

```

**Pull request merged**